Polyfills/Window/Source/Windows/WindowPlatform.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <Windows.h>
+#include <Babylon/JsRuntime.h>
+
+#include "..\Window.h"
+
+using WindowType = HWND;
+
+namespace Babylon::Polyfills::Internal
+{
+    Napi::Value Window::GetDevicePixelRatio(const Napi::CallbackInfo& info)
+    {
+        auto& window = *static_cast<Window*>(info.Data());
+
+        UINT dpi = GetDpiForWindow(*((HWND*)(window.m_windowPtr)));
+        return Napi::Value::From(info.Env(), (float)dpi / 96.0f);
```

Comment:
For BabylonReactNative windowPtr is the IUknown/abi pointer for a SwapChainPanel not a hwnd. I believe this cast will generate bad memory/an eventual access violation

Reply:
Hmm, that definitely complicates things for supporting Win32 and UWP. I suppose i should have separate implementations for both, enforced with some preprocessor directive? are there any preprocessor variables for UWP?

Reply:
I'm not aware of a specific C++ macro for that, but you could have different implementation files for Win32 and UWP include them separately in the CMake configuration.

That said, in the case @chrisfromwork mentioned, we probably need the ability to specify the scale factor externally from the BabylonReactNative component. For UWP where we're backed by a `SwapChainPanel`, we'd presumably want to use `SwapChainPanel.CompositionScaleX` and `SwapChainPanel.CompositionScaleY` as the scaling ratio to correctly integrate with XAML's rendering.

Reply:
this isn't strictly a uwp vs win32 distinction. BabylonNative UWP flavors use a core window pointer for the window ptr:
https://github.com/BabylonJS/BabylonNative/blob/93005ea71a30f489eac3ad1d98878ac50532b51a/Apps/Playground/UWP/App.cpp#L146

BabylonReactNative UWP flavors use a swap chain panel for the window ptr:
https://github.com/BabylonJS/BabylonReactNative/blob/95ecb8305edbe58f73f3a49e743d8b7ea428f8c5/Modules/%40babylonjs/react-native/windows/BabylonReactNative/EngineView.cpp#L35

it might be more practical to hand in a std::function that returns the scale factor needed compared to trying to determine what to do behind the scenes with the void* window ptr.

Reply:
If I recall correctly, the idea behind having the `windowPtr` be templated was that in the future we'd have template specializations per platform that would make it more type safe (rather than just being `void*`). If we actually implemented that now, then it seems like the template specializations for each type (including `ICoreWindow` vs `SwapChainPanel`) can take care of ensuring we are doing the right thing to get the pixel scaling value. @syntheticmagus, @bghgary - any comments on the window type template specialization?

Reply:
What @ryantrem said sound right to me. It should depend on the specializations that are allowed for the platform. For Win32, this is probably an HWND only. For UWP, this can either be a CoreWindow to take over the whole window or a SwapChainPanel. At some point, we may add support for exclusive fullscreen if bgfx adds support for it which may be a completely different parameter.

The switching between platforms should end up in the CMake step as much as possible. We should try to avoid using macros to switch.

Reply:
I think since `Graphics` is where we will have the template specializations for window type, and since that is where we have hardware scaling level, and since using the `window` polyfill is optional, and finally since ideally this value would be accessible to native code as well as JavaScript, the core implementation of `GetDevicePixelRatio` should be implemented on and publicly exposed from `Graphics`, and then the `window` polyfill should just expose that value to JavaScript.

Reply:
This implementation makes more sense, now the polyfill is just how we expose this functionality to Javascript, while we keep our Graphics::Impl object as our source of truth and where we should handle these system calls. The latest revision should reflect this change.